DZ_12/task_2.py

```diff
@@ -0,0 +1,22 @@
+#2.(необов'язкове виконання) Написати власний менеджер контексту, задачею якого буде
+# друкувати "==========" – 10 знаків дорівнює перед виконанням коду та після виконання коду,
+# таким чином виділяючи блок коду символами дорівнює.
+# У випадку виникнення будь-якої помилки вона має бути надрукована текстом,
+# проте програма не має завершити своєї роботи.
+
+
+class MyContextManager:
+    def __enter__(self):
+        print("=" * 10)
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        print("=" * 10)
+        if exc_type is not None:  # якщо є помилки, то вони будуть надруковані. Я перевірила і змінила "r" на "а"
+            print(f"Error: {exc_type} - {exc_value}") # /але якщо написати неіснуючий файл - дає помилку
+        return True     # FileNotFoundError(( треба розширювати контексний менеджер, але я ще добре не розумію як
```

Comment:
If an exception is supplied, and the method wishes to suppress the exception (i.e., prevent it from being propagated), it should return a true value. Otherwise, the exception will be processed normally upon exit from this method.